src/test/java/org/opensearch/security/tools/democonfig/InstallerTests.java

```diff
@@ -0,0 +1,512 @@
+import static org.opensearch.security.tools.democonfig.Installer.environment;
+import static org.opensearch.security.tools.democonfig.Installer.finishScriptExecution;
+import static org.opensearch.security.tools.democonfig.Installer.gatherUserInputs;
+import static org.opensearch.security.tools.democonfig.Installer.initializeVariables;
+import static org.opensearch.security.tools.democonfig.Installer.initsecurity;
+import static org.opensearch.security.tools.democonfig.Installer.printScriptHeaders;
+import static org.opensearch.security.tools.democonfig.Installer.printVariables;
+import static org.opensearch.security.tools.democonfig.Installer.readOptions;
+import static org.opensearch.security.tools.democonfig.Installer.resetState;
+import static org.opensearch.security.tools.democonfig.Installer.setBaseDir;
+import static org.opensearch.security.tools.democonfig.Installer.setOpenSearchVariables;
+import static org.opensearch.security.tools.democonfig.Installer.setSecurityVariables;
+import static org.opensearch.security.tools.democonfig.Installer.skip_updates;
+import static org.opensearch.security.tools.democonfig.util.DemoConfigHelperUtil.createDirectory;
+import static org.opensearch.security.tools.democonfig.util.DemoConfigHelperUtil.createFile;
+import static org.opensearch.security.tools.democonfig.util.DemoConfigHelperUtil.deleteDirectoryRecursive;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+
+public class InstallerTests {
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+    private final InputStream originalIn = System.in;
+
+    @Before
+    public void setUpStreams() {
+        System.setOut(new PrintStream(outContent));
+        resetState();
+    }
+
+    @After
+    public void restoreStreams() {
+        System.setOut(originalOut);
+        System.setIn(originalIn);
+    }
+
+    @Test
+    public void testPrintScriptHeaders() {
+        printScriptHeaders();
+
+        String expectedOutput = "### OpenSearch Security Demo Installer" + System.lineSeparator()
+            + "### ** Warning: Do not use on production or public reachable systems **" + System.lineSeparator();
+        assertThat(expectedOutput, equalTo(outContent.toString()));
+    }
+
+    @Test
+    public void testReadOptions_withoutHelpOption() {
+        // All options except Help `-h`
+        String[] validOptions = { "/scriptDir", "-y", "-i", "-c", "-s", "-t" };
+        readOptions(validOptions);
+
+        assertEquals("/scriptDir", SCRIPT_DIR);
+        assertThat(assumeyes, is(true));
+        assertThat(initsecurity, is(true));
+        assertThat(cluster_mode, is(true));
+        assertEquals(0, skip_updates);
+        assertEquals(ExecutionEnvironment.TEST, environment);
+    }
+
+    @Test
+    public void testReadOptions_help() {
```

Comment:
every time we add/change help text a unit test should break?  Can we verify this checking structured output instead of the process output?

Reply:
I'm moving this discussion to #3807 and resolving it here.